aten/src/ATen/native/Sorting.cpp

```diff
@@ -186,22 +197,38 @@ void quantile_impl(
         at::broadcast_tensors({q * last_index, sorted.isnan().any(-1, true)});
     ranks = at::masked_fill(tl[0], tl[1], last_index);
   }
-  Tensor ranks_below = ranks.toType(kLong);
-  Tensor weights = ranks - ranks_below;
-  Tensor ranks_above = ranks.ceil_().toType(kLong);
 
+  // adjust ranks based on the interpolation mode
+  if (interpolation == "lower") {
```

Comment:
FYI, NumPy's interpolation mode implementations can be found here:

https://github.com/numpy/numpy/blob/f4aadabcc0e1968f3b309642af52ef11ae77ca01/numpy/lib/function_base.py#L4004

Reply:
Thank you for the reference. We did follow this but we optimized the cases where we can avoid the interpolation.